core/iwasm/libraries/lib-socket/test/nslookup.c

```diff
@@ -0,0 +1,16 @@
+#include <assert.h>
+#include <stdio.h>
+#include <string.h>
+#ifdef __wasi__
+#include <wasi_socket_ext.h>
+#else
+#include <netdb.h>
+#endif
+
+int
+main()
```

Comment:
I'm not sure what exactly are we testing here? I'd suggest we test the result of `getaddrinfo`. Unfortunately, we can't control the responses (unless we mock DNS), but I think we can still test if there's any response or whether the filtering (`hints` parameter) works. What do you think?

Reply:
Yes, makes sense - updating!